src/ripple/app/tx/impl/PayChan.cpp

```diff
@@ -457,9 +461,22 @@ PayChanClaim::doApply()
         if (!sled)
             return terNO_ACCOUNT;
 
-        if (txAccount == src && ((*sled)[sfFlags] & lsfDisallowXRP))
+        // Obeying the lsfDisallowXRP flag was a bug.  Piggyback on
+        // featureDepositAuth to remove the bug.
+        bool const depositAuth {ctx_.view().rules().enabled(featureDepositAuth)};
+        if (!depositAuth &&
+            (txAccount == src && (sled->getFlags() & lsfDisallowXRP)))
             return tecNO_TARGET;
 
+        // Check whether the destination account requires deposit authorization
+        if (txAccount != dst)
+        {
+            // If dst set lsfDepositAuth and signer is not dst, then fail.
```

Comment:
Nit: I think we're better off leaving comments such as this out out of the code base. It's a comment that explains the one line of code below it, and that one line is already straight forward. I don't care about this particular comment, but it's something I noticed in a couple places in this PR.